### DISCUSSION TITLE
database structure update

### DISCUSSION THREAD

**Discussion by pgswow** [pocketbase/pocketbase#1247] [Q&A] [answered]
Hello, I did not find a similar question. I have 2 servers, a development server and a production server, when I update the database structure on the development server (add a new schema, change an existing one, add new table fields), how can I transfer the changes to the production database without losing the production database data?

**Comment by ganigeorgiev** [comment, marked as answer]
You can create a migration.

With v0.9+ there is support for "automigration" that will create a `pb_migrations` directory that you can commit safely to your repo. The migration files can be both Go or JS (for the latter documentation will be available soon). Migrations are automatically applied on `serve`.

You can check https://pocketbase.io/docs/migrations/ for more info on how to write migrations/enable automigrate when extending PocketBase.

**Reply by pgswow** [reply]
Thanks, here's what I tried. I updated both servers to 9.0.2, on the first server where there are changes I made a migration:

pocketbase.exe migrate create update_01
? Do you really want to create migration "pb_migrations/1671008144_update_01.js"? Yes
Successfully created file "pb_migrations/1671008144_update_01.js"

sent the directory with the migration to the second server and applied it "pocketbase migrate up", started the server and unfortunately did not see any changes =(

Am I doing everything right?

**Reply by ganigeorgiev** [reply]
I'm confused. I guess you are using the prebuilt binaries, right?

Do you have automigrate enabled? If yes, then when making collection changes from the UI, a migration file(s) will be created for you and they will be applied automatically on server start.

What is the content of your manually created migration?
`migrate create` creates a **blank** migration that you have to fill with your specific changes manually.

You can use `migrate collections` to create a full snapshot of your current collections configuration.

**Reply by pgswow** [reply]
Yes, i using the prebuilt binaries.

where i have enable automigrate? i not see this options in UI.

**Reply by ganigeorgiev** [reply]
It should be already enabled by default. If not (it is possible that on windows the tmp dir check may not be reliable), start the executable with `--automigrate=1`.

But in all cases you can trigger a full collections snapshot by running `migrate collections` (when automigrate is enabled for the first time it will actually create a full snapshot as a first migration to ensure that after this point a more granular migration files are created).

**Reply by ganigeorgiev** [reply]
@pgswow If you are still not able to apply the migrations on the prod server, you can:

- make sure that both environments has the same `pb_migrations` files content.
- run `sqlite3 pb_data/data.db 'select * from _migrations'` on both servers to check which migration files were applied

**Reply by pgswow** [reply]
> @pgswow If you are still not able to apply the migrations on the prod server, you can:
> 
> * make sure that both environments has the same `pb_migrations` files content.
> * run `sqlite3 pb_data/data.db 'select * from _migrations'` on both servers to check which migration files were applied

Thank you, it was not necessary, the migrations worked, everything is just fine, thank you very much! =)
